### DISCUSSION TITLE
Improving Creature Animation Browser

### DISCUSSION THREAD

**Discussion by Incrementis** [NearInfinityBrowser/NearInfinity#154] [Ideas & Feature Requests]
Hello,

This is a topic I had previously posted on the SHS forums, but when SHS was shut down my post disappeared as well.
I'm reposting this here because @Argent77 agreed to look into it.

Here it is (man, I'm so glad I'm taking notes):

![grafik](https://github.com/NearInfinityBrowser/NearInfinity/assets/17119474/3b28f1a1-a3b3-405d-8cb9-1d9be62ed028)
When I test armor and weapon colors, I open all these windows. Change the colors in the item ability effects, then save the item file and select it in the "_Creature Animation Browser_".

The problem here is that I have to press F5 to refresh the "_Creature Animation Browser_" to see the element's color changes. This closes all open NI windows. This is a very cumbersome and time-consuming process.

I assume that when you first open NI, it loads all game files once and saves the file contents in some kind of buffer.
It would be great if this could be improved, e.g. by pressing the "_APPLY_" button ensures that the item file is loaded again and not just once.

This is just a suggestion and I assume there are certainly many other ways to solve this problem.

Thank you for reading this comment!

**Comment by Argent77** [comment]
This issue is already on my TODO list. However, it's not trivial to fix, since creature equipment and other resources are cached to speed up loading times. I'll try to find a solution though.

**Comment by Argent77** [comment]
[This commit](https://github.com/Argent77/NearInfinity/commit/49f2a28e67090b39ab2b17ef76f5af5cef44d74d) *should* fix this issue. It will be available with the next public release.

**Reply by Incrementis** [reply]
Thank you for the effort.
I'm very excited to see how the result will feel, but seriously, don't put yourself under pressure. 
You are the expert here.

**Comment by Argent77** [comment]
Implemented in [v2.3-20230609](https://github.com/Argent77/NearInfinity/releases/tag/v2.3-20230609).
